other/travis/env-linux.sh

```diff
@@ -11,5 +11,8 @@ RUN() {
 }
 
 TESTS() {
-  "$@"
+  # Keep running tests until they eventually succeed or Travis times out after
+  # 50 minutes. This cuts down on the time lost when tests fail, because we no
+  # longer need to manually restart the build and wait for compilation.
+  "$@" || TESTS "$@"
```

Comment:
50 minutes is a bit too much imo. Maybe make a limit here on run it 3 times and then fail? If tests are broken this will take a lot of travis capacity that could be used on building other repos.